kolena/_experimental/dataset/evaluation.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2021-2024 Kolena Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from dataclasses import asdict
+from typing import List
+from typing import Optional
+
+import pandas as pd
+
+from kolena._api.v1.event import EventAPI
+from kolena._api.v2.model import LoadByTagRequest
+from kolena._api.v2.model import LoadByTagResponse
+from kolena._api.v2.model import Path
+from kolena._utils import krequests_v2 as krequests
+from kolena._utils.instrumentation import with_event
+from kolena._utils.serde import from_dict
+from kolena.dataset.evaluation import download_results
+from kolena.dataset.evaluation import EvalConfigResults
+from kolena.errors import IncorrectUsageError
+from kolena.errors import NotFoundError
+
+
+def _get_models_by_tag(tag: str) -> List[str]:
+    """
+    Get model names by a given tag.
+
+    :param tag: The tag on the model.
+    :return: A list of model names associated with the provided tag.
+    """
+    response = krequests.put(
+        Path.LOAD_BY_TAG,
+        json=asdict(LoadByTagRequest(tag=tag)),
+    )
+    response.raise_for_status()
+
+    load_by_tag_response = from_dict(LoadByTagResponse, response.json())
+    return [model.name for model in load_by_tag_response.models]
+
+
+@with_event(EventAPI.Event.FETCH_DATASET_MODEL_RESULT_BY_TAG)
+def download_results_by_tag(
+    dataset: str,
+    model_tag: str,
+    commit: Optional[str] = None,
+    include_extracted_properties: bool = False,
+) -> tuple[pd.DataFrame, list[EvalConfigResults]]:
```

Comment:
I don't think typing like `tuple` and `list` existed until 3.9, which means we need to use `typing.Tuple` and `typing.List` as long as we support 3.8.

https://peps.python.org/pep-0585/